### PR TITLE
Time step with predictor velocities

### PR DESCRIPTION
**Pull request by glwagner**
This PR refactors the time-stepping algorithm to use the "predictor velocities" abstraction in the fractional step method. This brings the implemented algorithm in line with the algorithm described in the documentation.

### CODE REVIEW

**Comment by glwagner** [comment]
@ali-ramadhan I may need some help with the checkpointer...

**Comment by ali-ramadhan** [comment]
I'll have a look

File: src/Solvers/solve_for_pressure.jl [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
         ϕ = RHS = solver.storage.storage1
     else
         ϕ = RHS = solver.storage
     end
 
     @launch(device(arch), config=launch_config(grid, :xyz),
-            calculate_pressure_right_hand_side!(RHS, solver.type, arch, grid, U, G, Δt))
+            calculate_pressure_right_hand_side!(RHS, solver.type, arch, grid, Δt, U★))
 
     solve_poisson_equation!(solver, grid)
```
Comment (ali-ramadhan) [inline src/Solvers/solve_for_pressure.jl]: Not sure why but one of the Travis builds hits a `ReadOnlyMemoryError` on this line and I get a segmentation fault on my computer at this line.

Perhaps part of the model is corrupted after restoring from checkpoint? I'll try to find out.

Travis build log: https://travis-ci.com/climate-machine/Oceananigans.jl/jobs/294185099#L453

**Comment by ali-ramadhan** [comment]
Think I fixed it. At least checkpointer tests pass on my laptop.

**Pull request merged**